src/state/user/updater.tsx

```diff
@@ -5,31 +5,65 @@ import { useInterval } from 'react-use'
 import { DEFAULT_SLIPPAGE, DEFAULT_SLIPPAGE_STABLE_PAIR_SWAP, MAX_NORMAL_SLIPPAGE_IN_BIPS } from 'constants/index'
 import { AppDispatch, AppState } from 'state/index'
 import { useCheckStablePairSwap } from 'state/swap/hooks'
-import { useUserSlippageTolerance } from 'state/user/hooks'
+import { usePoolSlippageTolerance, useUserSlippageTolerance } from 'state/user/hooks'
 
-import { updateUserDegenMode } from './actions'
+import { updatePoolDegenMode, updateUserDegenMode } from './actions'
 
 export default function Updater(): null {
   const dispatch = useDispatch<AppDispatch>()
-  const degenMode = useSelector<AppState, AppState['user']['userDegenMode']>(state => state.user.userDegenMode)
-  const userDegenModeAutoDisableTimestamp = useSelector<
+  const isStablePairSwap = useCheckStablePairSwap()
+
+  const swapDegenMode = useSelector<AppState, AppState['user']['userDegenMode']>(state => state.user.userDegenMode)
```

Comment:
because only 1 hook above, you cannot use the hook here to get the 2 variables/updaters at the same time.